operators/knative-istio-controller/src/charm.py

```diff
@@ -63,23 +65,25 @@ def render(self):
             plural="peerauthentications",
             verbs=None,
         )
+        try:
+            gateway_data = self.gateway_relation.get_relation_data()
+        except Exception as error:
```

Comment:
Catching a generic `Exception` is usually an antipattern, we should look here for specifc exceptions the library is raising.